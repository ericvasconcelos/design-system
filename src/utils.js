export { arrayCombinations } from 'utils/arrayCombinations'
export { getBackgroundColor } from 'utils/getBackgroundColor'
export { getBorder } from 'utils/getBorder'
export { getButtonStyle } from 'utils/getButtonStyle'
export { getBorderRadius } from 'utils/getBorderRadius'
export { getColorFromName } from 'utils/getColorFromName'
export { getGuttersNames } from 'utils/getGuttersNames'
export { getIconNames } from 'utils/getIconNames'
export { getPadding } from 'utils/getPadding'
export { getPaddingNames } from 'utils/getPaddingNames'
export { getTypography } from 'utils/getTypography'
export { parsePalleteName } from 'utils/parsePalleteName'
export { styleColorValidator } from 'utils/styleColorValidator'
export { useAriaProps } from 'utils/useAriaProps'

export { DateUtils } from 'utils/DateUtils'
export { QueryUtils } from 'utils/QueryUtils'
export { ObjectUtils }  from 'utils/ObjectUtils'
export { CookiesUtils } from 'utils/CookiesUtils'
export { MoneyUtils } from 'utils/MoneyUtils'
export { NumberUtils } from 'utils/NumberUtils'