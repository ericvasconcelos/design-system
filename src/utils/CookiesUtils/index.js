export class CookiesUtils {
  static toObject(string = '') {
    const cookies = {}
    if (string !== '') {
      const parts = string.split('; ')
      parts.forEach(part => {
        const [key, value] = part.split('=')
        cookies[key] = value
      })
    }
    return cookies
  }
}
