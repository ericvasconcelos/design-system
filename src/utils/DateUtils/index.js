export class DateUtils {
  static toDate(string, format = 'pt-br') {
    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = string?.split('/')
        break
      case 'iso':
        [year, month, day] = string?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    return new Date(year, parseInt(month, 10) - 1, day)
  }

  static getAge(date) {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const difference = Date.now() - date.getTime()

    return Math.abs(new Date(difference).getUTCFullYear() - 1970)
  }

  static toString(date, format = 'pt-br') {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
    const month = date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1
    const year = date.getFullYear()

    switch (format) {
      case 'pt-br':
        return `${day}/${month}/${year}`
      case 'iso':
        return `${year}-${month}-${day}`
      default:
        throw new Error('Format invalid')
    }
  }

  static validate(date, format = 'pt-br') {
    const monthDaysMap = {
      '01': 31,
      '02': 29,
      '03': 31,
      '04': 30,
      '05': 31,
      '06': 30,
      '07': 31,
      '08': 31,
      '09': 30,
      10: 31,
      11: 30,
      12: 31
    }
    let isValid = false

    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = date?.split('/')
        break
      case 'iso':
        [year, month, day] = date?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    isValid = isValid || parseInt(day, 10) <= monthDaysMap[month]
    isValid = isValid && parseInt(year, 10) > 1900

    return isValid
  }

  static convertMonthToYear(month) {
    return parseInt(month, 10) / 12
  }

  static formatPeriod(period, periodType) {
    const periodTypeMap = {
      monthly: { singular: 'mÃªs', plural: 'meses' },
      yearly: { singular: 'ano', plural: 'anos' },
      daily: { singular: 'dia', plural: 'dias' }
    }
    return `${period} ${periodTypeMap[periodType][period < 2 ? 'singular' : 'plural']}`
  }
}
