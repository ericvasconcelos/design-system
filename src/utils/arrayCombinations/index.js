export const arrayCombinations = (...args) => {
  const returnable = []
  const max = args.length - 1
  function helper(arr, i) {
    for (let j = 0, { length } = args[i]; j < length; j++) {
      const newArray = arr.slice(0) // clone arr
      newArray.push(args[i][j])
      if (i === max) returnable.push(newArray)
      else helper(newArray, i + 1)
    }
  }
  helper([], 0)
  return returnable
}
