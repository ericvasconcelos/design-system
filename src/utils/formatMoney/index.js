import { toMoney, toNumber } from 'vanilla-masker'

export const formatMoney = entry => {
  const moneyFormatter = {
    precision: 2,
    separator: ',',
    delimiter: '.',
    unit: 'R$'
  }

  return toMoney(toNumber(entry), moneyFormatter)
}
