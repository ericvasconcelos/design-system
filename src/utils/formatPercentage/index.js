import { toMoney, toNumber } from 'vanilla-masker'

export const formatPercentage = entry => {
  const percentageFormatter = {
    precision: 1,
    separator: ',',
    suffixUnit: '%'
  }
  return toMoney(toNumber(entry), percentageFormatter)
}
