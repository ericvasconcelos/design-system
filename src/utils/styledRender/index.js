import React from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from 'styles/theme'

export const styledRender = (children, options = {}) => {
  const rendered = render(<ThemeProvider theme={theme}>{children}</ThemeProvider>, options)
  return {
    ...rendered,
    rerender: (ui, rerenderOptions = {}) =>
      styledRender(ui, { container: rendered.container, ...rerenderOptions })
  }
}
