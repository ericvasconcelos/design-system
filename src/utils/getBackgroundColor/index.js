import { getColorFromName } from 'utils/getColorFromName'

export const getBackgroundColor = ({ theme, styleColor }) =>
  `background-color: ${getColorFromName({ theme, styleColor })};`
