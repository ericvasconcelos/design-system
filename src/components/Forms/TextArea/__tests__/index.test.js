import React from 'react'

import userEvent from '@testing-library/user-event'
import { Formik, Form } from 'formik'
import { styledRender } from 'utils/styledRender'

import { TextAreaBasic, TextAreaWithFormik } from '../index.js'

describe('Should render TextAreaBasic when', () => {
  it('has default props', () => {
    const { getByTestId, getByText } = styledRender(
      <TextAreaBasic
        id="testId"
        name="testName"
        label="some label here"
        informativeMessage="some informative message here"
      />
    )

    userEvent.type(getByTestId('TextAreaBasic'), 'some content typed here')

    expect(getByText('some label here')).toBeInTheDocument()
    expect(getByTestId('TextAreaBasic')).toMatchSnapshot()
  })

  it('has error status', () => {
    const { getByTestId, getByText } = styledRender(
      <TextAreaBasic
        id="testId"
        name="testName"
        label="some label here"
        informativeMessage="some informative message here"
        errorMessage="some error message here"
        status="error"
      />
    )

    expect(getByText('some error message here')).toBeInTheDocument()
    expect(getByTestId('TextAreaBasic')).toMatchSnapshot()
  })

  it('has default props using Formik', () => {
    const { getByTestId, getByText } = styledRender(
      <Formik initialValues={{ name: 'textArea' }}>
        <Form>
          <TextAreaWithFormik
            id="testId"
            name="testName"
            label="some label here"
            informativeMessage="some informative message here"
          />
        </Form>
      </Formik>
    )

    userEvent.type(getByTestId('TextAreaBasic'), 'some content typed here')

    expect(getByText('some label here')).toBeInTheDocument()
    expect(getByTestId('TextAreaBasic')).toMatchSnapshot()
  })
})
