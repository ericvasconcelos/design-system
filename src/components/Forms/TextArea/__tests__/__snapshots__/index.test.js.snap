// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TextAreaBasic when has default props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: content-box;
  width: calc(100% - 2 * 1rem);
  border-radius: 8px;
  padding: 0.625rem 0.9375rem;
  background-color: #FFFFFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-shadow: 0px 0px 0px 3px transparent;
  border: 1px solid;
  border-color: #A6B5BB;
}

.c2 {
  color: #69788C;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1.056rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  top: unset;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  line-height: 1.0625rem;
  text-align: left;
}

.c4 {
  color: #00224B;
  font-size: 1.0625rem;
  font-weight: 400;
  line-height: 1.381rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  display: block;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  line-height: 1.375rem;
  text-align: left;
}

.c4:-webkit-autofill,
.c4:-internal-autofill-selected {
  -webkit-animation-name: onAutoFillStart;
  animation-name: onAutoFillStart;
  -webkit-text-fill-color: #455A73;
  box-shadow: inset 0 0 0 100em #FFFFFF;
}

.c4:not(:-webkit-autofill) {
  -webkit-animation-name: onAutoFillCancel;
  animation-name: onAutoFillCancel;
}

.c3 {
  color: #455A73;
}

.c1 {
  width: calc(100% - 2 * 1.5rem);
  padding: 1.5rem;
  height: auto;
}

.c5 {
  height: 6rem;
  min-height: 6rem;
  width: 100%;
  overflow: auto;
  padding: 0;
}

<div
  class="c0 c1"
  data-testid="TextAreaBasic"
  for="testId"
>
  <span
    class="c2 c3"
    id="labeltestId"
  >
    some label here
  </span>
  <span
    aria-describedby="infoMessagetestId"
    aria-labelledby="testId"
    aria-multiline="true"
    class="c4 c5"
    contenteditable="true"
    id="testId"
    name="testName"
    role="textbox"
    rows="3"
    value=""
  />
</div>
`;

exports[`Should render TextAreaBasic when has default props using Formik 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: content-box;
  width: calc(100% - 2 * 1rem);
  border-radius: 8px;
  padding: 0.625rem 0.9375rem;
  background-color: #FFFFFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-shadow: 0px 0px 0px 3px transparent;
  border: 1px solid;
  border-color: #A6B5BB;
}

.c2 {
  color: #69788C;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1.056rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  top: unset;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  line-height: 1.0625rem;
  text-align: left;
}

.c4 {
  color: #00224B;
  font-size: 1.0625rem;
  font-weight: 400;
  line-height: 1.381rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  display: block;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  line-height: 1.375rem;
  text-align: left;
}

.c4:-webkit-autofill,
.c4:-internal-autofill-selected {
  -webkit-animation-name: onAutoFillStart;
  animation-name: onAutoFillStart;
  -webkit-text-fill-color: #455A73;
  box-shadow: inset 0 0 0 100em #FFFFFF;
}

.c4:not(:-webkit-autofill) {
  -webkit-animation-name: onAutoFillCancel;
  animation-name: onAutoFillCancel;
}

.c3 {
  color: #455A73;
}

.c1 {
  width: calc(100% - 2 * 1.5rem);
  padding: 1.5rem;
  height: auto;
}

.c5 {
  height: 6rem;
  min-height: 6rem;
  width: 100%;
  overflow: auto;
  padding: 0;
}

<div
  class="c0 c1"
  data-testid="TextAreaBasic"
  for="testId"
>
  <span
    class="c2 c3"
    id="labeltestId"
  >
    some label here
  </span>
  <span
    aria-describedby="infoMessagetestId"
    aria-labelledby="testId"
    aria-multiline="true"
    class="c4 c5"
    contenteditable="true"
    id="testId"
    name="testName"
    role="textbox"
    rows="3"
    value=""
  />
</div>
`;

exports[`Should render TextAreaBasic when has error status 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: content-box;
  width: calc(100% - 2 * 1rem);
  border-radius: 8px;
  padding: 0.625rem 0.9375rem;
  background-color: #FFFFFF;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-shadow: 0px 0px 0px 3px #D93E3E3D;
  border: 1px solid;
  border-color: #D93E3EA3;
}

.c2 {
  color: #B33333;
  font-size: 0.8125rem;
  font-weight: 500;
  line-height: 1.056rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  top: unset;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  line-height: 1.0625rem;
  text-align: left;
}

.c4 {
  color: #00224B;
  font-size: 1.0625rem;
  font-weight: 400;
  line-height: 1.381rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  display: block;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  line-height: 1.375rem;
  text-align: left;
}

.c4:-webkit-autofill,
.c4:-internal-autofill-selected {
  -webkit-animation-name: onAutoFillStart;
  animation-name: onAutoFillStart;
  -webkit-text-fill-color: #455A73;
  box-shadow: inset 0 0 0 100em #FFFFFF;
}

.c4:not(:-webkit-autofill) {
  -webkit-animation-name: onAutoFillCancel;
  animation-name: onAutoFillCancel;
}

.c3 {
  color: #455A73;
}

.c1 {
  width: calc(100% - 2 * 1.5rem);
  padding: 1.5rem;
  height: auto;
}

.c5 {
  height: 6rem;
  min-height: 6rem;
  width: 100%;
  overflow: auto;
  padding: 0;
}

<div
  class="c0 c1"
  data-testid="TextAreaBasic"
  for="testId"
>
  <span
    class="c2 c3"
    id="labeltestId"
  >
    some label here
  </span>
  <span
    aria-describedby="infoMessagetestId"
    aria-labelledby="testId"
    aria-multiline="true"
    class="c4 c5"
    contenteditable="true"
    id="testId"
    name="testName"
    role="textbox"
    rows="3"
    value=""
  />
</div>
`;
